assignment/index.html

```diff
@@ -57,29 +57,62 @@
        Remember: functions are meant to facilitate code comprehension and reuse. Try
        your best to use them to organize your code.
       ===================== */
-
-
-      /* =====================
-
-      Start code
-
-      ===================== */
-
-      var jsonToCsv = function(json) { console.log(json); };
-
-
-      var addMarkers = function(map) {};
-
-      /* =====================
-
-      End code
-
-      ===================== */
-
-
-      // `healthCenters` is defined in health_centers.js
-      jsonToCsv(healthCenters);
-      addMarkers(map);
+      
+    var jsonToCsv = function(json) { 
+        console.log(json); 
+        let a = json;
+        //Initalize the array that will contain the information in the JSON file
+        var arr = [['LNG','LAT','OBJECTID','NAME','ORGANIZATION','ZIP','PHONE','WEBSITE','DENTAL_PHONE','FULL_ADDRESS'],[],[],[],[],[],[],[],[],[],[]];
+        //Iterate through the JSON file and push the elements to the respective array 
+        for(i=0;i<a.length;i++){
+          //console.log(a[i]);
+          arr[1].push(a[i].LNG);
+          arr[2].push(a[i].LAT);
+          arr[3].push(a[i].OBJECTID);
+          arr[4].push(a[i].NAME);
+          arr[5].push(a[i].ORGANIZATION);
+          arr[6].push(a[i].ZIP);
+          arr[7].push(a[i].PHONE);
+          arr[8].push(a[i].WEBSITE);
+          arr[9].push(a[i].DENTAL_PHONE);
+          arr[10].push(a[i].FULL_ADDRESS);
+        }
```

Comment:
You did this in a slightly unorthodox way (I'm not going to say it was wrong because I think there's a good argument to be made for representing a CSV just like this).

As for performance: I don't think you should sweat it and I don't think there is much room for improvement. Luckily for us, of the actual interpreter which underwrites modern javascript evaluation is highly optimized down to C (that's what Google's/node.js' V8 engine does).

As for readability, you might prefer something like this (and we might even want to dispense with the loops for readability's sake... more on that in a coming class):
```javascript
// our header is here
var keys = Object.keys(a)
// create the array of arrays for the body
var body = []
// a bit of setup in anticipation of the loop ahead
for (key in keys) { body.push([]) }
for (var k = 0; k < keys.length in keys; k = k + 1) {
  for (var i = 0; i < a[i].length; i = i + 1) {
    var key = keys[k]
    body[k] = a[i][key]
  }
}
var result = [keys].concat(body)
```

Reply:
Assuming this code works (pretty sure, but I typed it in haste and didn't test it), it should generalize to key-value mappings regardless of the underlying keys or their number